asclepias_broker/harvester/receivers.py

```diff
@@ -12,15 +12,18 @@
 
 def harvest_metadata_after_event_process(app, event: Event = None):
     """."""
-    identifiers = set()
-    identifier_events = (obj_event for obj_event in event.object_events
-                         if obj_event.payload_type == PayloadType.Identifier)
+    identifiers = list()
+    identifier_events = set(obj_event for obj_event in event.object_events
```

Comment:
I'm not 100% sure if `obj_event` (which is a `.events.models.EventObject`) is hashable and thus making a set of these might not work as expected. For that reason, it's better also to keep hashable primitive values (`str`, `tuple`, `frozenset`, etc) around, and serialize to a `list` of `dict`s before passing to `current_harvester.publish_metadata_harvest`:

```python
# Build a set of tuple(identifier, scheme, frozenset(providers))
identifiers = {
    ('10.1234/1', 'doi', frozenset(['Provider A', 'Provider B', ...])),
    ('10.1234/2', 'doi', frozenset(['Provider A', 'Provider B', ...])),
    ...
}

payloads = [
    dict(identifier=identifier, scheme=scheme, providers=list(providers)) 
    for identifier, scheme, providers in identifiers
]
current_harvester.publish_metadata_harvest(payloads)
```